src/sql/parts/objectExplorer/viewlet/serverTreeActionProvider.ts

```diff
@@ -181,7 +181,10 @@ export class ServerTreeActionProvider extends ContributableActionProvider {
 
 		this.addScriptingActions(context, actions);
 
-		if (isAvailableDatabaseNode) {
+		let serverInfo = this._connectionManagementService.getServerInfo(context.profile.id);
+		let isCloud = (serverInfo === undefined || serverInfo === null)? false : serverInfo.isCloud;
```

Comment:
you could replace `(serverInfo === undefined || serverInfo === null)` with `!serverInfo`, which I think is a bit "cleaner", IMO.

Also, can `context.profile` ever be null in the above line?

Reply:
profiler.* properties are being used anyways before this - for addQueryActions etc - I am depending on that. I could add an explicit check but it would have failed before anyways.

Reply:
Not sure why its not showing the reply :) Pasting again 

profiler.* properties are being used before this - for addQueryActions etc - I am depending on that. I could add an explicit check but it would have failed before anyways.